### PR TITLE
Allow starting a number with a decimal point

### PR DESCRIPTION
**Pull request by mhmdanas**
Fixes https://github.com/sharkdp/insect/issues/221.

### CODE REVIEW

File: src/Insect/Parser.purs [hunk 1]
```diff
@@ -89,10 +89,10 @@ whiteSpace = token.whiteSpace
 -- | Parse a number.
 number ∷ P Decimal
 number = do
-  intPart ← digits
-
-  mFracPart ← optionMaybe (append <$> string "." <*> digits)
-  let fracPart = fromMaybe "" mFracPart
+  decimalPart ← fractionalPart <|> do
```
Comment (mhmdanas) [inline src/Insect/Parser.purs:92]: Suggestions on how to better implement this are welcome, I feel like I could do this better.

File: test/Main.purs [hunk 2]
```diff
@@ -110,11 +110,17 @@ main = runTest do
         , "+3.50"
         ]
 
+      allParseAs (Expression (scalar 0.2))
+        [ "0.2"
+        , "  0.2  "
+        , "+0.2 "
+        , ".2"
+        ]
```
Comment (sharkdp) [inline test/Main.purs:118]: Before I take a closer look at the implementation: can we think edge cases that we need to test? For now, we have just one test for the new syntax (`".2"`). Do we also want to test things like `"+.2"`, `"-.2"`, `".02"` and add a few things that we *don't* want to be parsed as a number (e.g. `". 2"`)?

**Review by sharkdp — COMMENTED** [review]
Awesome! Thank you very much.

**Review by sharkdp — APPROVED** [review]
Thank you for the update!

**Comment by sharkdp** [comment]
Deployed on insect.sh/
